clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -457,6 +459,14 @@ public static ByteBuffer toBuffer(Struct struct) {
         return tps;
     }
 
+    @SuppressWarnings("unchecked")
+    public static <T extends Throwable> T assertFutureThrows(KafkaFuture<?> future, Class<T> exceptionClass) {
+        ExecutionException exception = assertThrows(ExecutionException.class, () -> future.get());
+        assertTrue("Unexpected exception cause " + exception.getCause(),
+                exceptionClass.isAssignableFrom(exception.getCause().getClass()));
+        return (T) exception.getCause();
```

Comment:
I think you can avoid the warning (and suppression) if you use `exceptionClass.cast`.

Reply:
Woah. Didn't know about that.